Google.PowerShell/Common/ProcessUtils.cs

```diff
@@ -0,0 +1,185 @@
+﻿// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Google.PowerShell.Common
+{
+    /// <summary>
+    /// This class contains the output of a running process.
+    /// </summary>
+    public sealed class ProcessOutput
+    {
+        /// <summary>
+        /// Whether the process succeeded or not.
+        /// </summary>
+        public bool Succeeded { get; }
+
+        /// <summary>
+        /// The complete contents of the stderr stream.
+        /// </summary>
+        public string StandardError { get; }
+
+        /// <summary>
+        /// The complete contents of the stdout stream.
+        /// </summary>
+        public string StandardOutput { get; }
+
+        public ProcessOutput(bool succeeded, string standardOutput, string standardError)
+        {
+            Succeeded = succeeded;
+            StandardOutput = standardOutput;
+            StandardError = standardError;
+        }
+    }
+
+    /// <summary>
+    /// The output streams from a process.
+    /// </summary>
+    public enum OutputStream
+    {
+        None,
+        StandardError,
+        StandardOutput
+    }
+
+    /// <summary>
+    /// Evemt args passed to the output handler of a process.
+    /// </summary>
+    public class OutputHandlerEventArgs : EventArgs
+    {
+        public string Line { get; }
+        public OutputStream OutputStream { get; }
+
+        public OutputHandlerEventArgs(string line, OutputStream stream)
+        {
+            Line = line;
+            OutputStream = stream;
+        }
+    }
+
+    /// <summary>
+    /// This class defines helper methods for starting sub-processes and getting the output from
+    /// the processes, including a helper to parse the output as json.
+    /// </summary>
+    public static class ProcessUtils
```

Comment:
Having written similar types of wrappers in the past, can we add some unit tests if possible? Even if the command is "echo %TMP%", it would be useful to have non-zero coverage of this tricky-to-get-right code.